### PR TITLE
SG-35291 Update cryptography to 43.0.1

### PR DESCRIPTION
**Pull request by carlos-villavicencio-adsk**
- Updates cryptography to 43.0.1
- Updates pyOpenSSL to 24.2.1 (required for cryptography)
- Fixes CVE-2024-6119
- Fixes CVE-2024-4603
- Fixes CVE-2024-2511
- Updates .snyk file
- Closes #238 
- Closes #239
- Closes #240 
- Closes #259 
- Closes #260 
- Closes #261 
- Closes #238 
- Azure Pipelines improvements

### CODE REVIEW

File: .snyk [hunk 1]
```diff
@@ -2,6 +2,10 @@
 version: v1.25.0
 ignore:
   SNYK-PYTHON-SETUPTOOLS-3180412:
+    - resources/python/requirements/3.7/requirements.txt
+    - resources/python/requirements/3.9/requirements.txt
+    - resources/python/requirements/3.10/requirements.txt
+    - resources/python/requirements/3.11/requirements.txt
```
Comment (julien-lang) [inline .snyk:8]: This does not sound like a good idea. Why this change?

Comment (carlos-villavicencio-adsk) [reply .snyk:8]: This only applies to ignore setuptools false positive ID `SNYK-PYTHON-SETUPTOOLS-3180412`.

Comment (carlos-villavicencio-adsk) [reply .snyk:8]: @julien-lang SG-31202

Comment (julien-lang) [reply .snyk:8]: Can you add this ticket ID in the YML as comment please?

Then, I can approve the PR.

File: resources/python/pipelines/pipelines.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
 - job: install_source_dependencies_3_9
   displayName: Install source dependencies Python 3.9
   dependsOn: install_source_dependencies_3_7
-  condition: and(not(endsWith( variables['System.PullRequest.SourceBranch'], '-automated')), ne(variables['Build.SourceBranch'], 'refs/heads/master'), not(startsWith(variables['Build.SourceBranch'], 'refs/tags/v')))
+  condition: and(succeeded(), not(endsWith( variables['System.PullRequest.SourceBranch'], '-automated')), ne(variables['Build.SourceBranch'], 'refs/heads/master'), not(startsWith(variables['Build.SourceBranch'], 'refs/tags/v')))
```
Comment (julien-lang) [inline resources/python/pipelines/pipelines.yml:40]: Can you tell me more about these changes?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -38,10 +38,6 @@ jobs:
       # Required when binary dependencies are not bundled
       - attrs==22.2.0    # Fix version. Otherwise tk-ci-tools will install latest
       - Twisted==22.10.0 # We use 22.10.0 to avoid breaking CI for Python 3.7
-      - certifi==2024.7.4
-      - autobahn==22.12.1
-      - pyOpenSSL==24.0.0
-      - service-identity==21.1.0
```
Comment (julien-lang) [inline azure-pipelines.yml:44]: Why are we free to clear these ones?

Comment (carlos-villavicencio-adsk) [reply azure-pipelines.yml:44]: They were added so these dependencies can be found by Python environment if bundles were not present.

Introduced when we add Python 3.10 but not binaries for 3.10. Later on, we added 3.10 binaries and we introduced 3.11 but no binaries for it. Now we have both CI running up to 3.11 and bundled binaries.

**Review by julien-lang — APPROVED** [review]

**Review by eduardoChaucaGallegos — APPROVED** [review]

**Pull request merged**